spec/cucumber/events/bus_spec.rb

```diff
@@ -1,94 +1,185 @@
-require "cucumber/events/bus"
+require 'cucumber/events/bus'
 
-module Cucumber
-  module Events
-    class TestEvent
+describe Cucumber::Events::Bus do
+  subject(:bus) { described_class.new(name_resolver) }
+
+  let(:name_resolver) { instance_double('Cucumber::Events::NameResolver') }
+
+  let!(:test_event_klass) do
+    class_double('Events::TestEvent').tap do |double|
+      stub_const('Events::TestEvent', double)
+      allow(double).to receive(:new).and_return(test_event_instance)
+      allow(double).to receive(:to_s).and_return('Events::TestEvent')
+      allow(double).to receive(:is_a?).with(Class).and_return(true)
     end
+  end
```

Comment:
I'd at least use an inline call to `Class.new` rather than all this stubbing stuff. But honestly for me the trade-off with polluting the namespace is not worth it.